.github/workflows/shared-java-publish-versioned-package.yaml

```diff
@@ -30,38 +33,39 @@ jobs:
           distribution: 'temurin'
           java-version: ${{ inputs.java_version }}
 
-      - name: Download Key
+      - name: Download key
         uses: IABTechLab/uid2-shared-actions/actions/download_gpg_key@main
         with:
           key: ${{ secrets.GPG_KEY }}
 
-      - name: Local vulnerability scanner 
+      - name: Generate Trivy vulnerability scan report
         uses: aquasecurity/trivy-action@master
+        if: inputs.publish_vulnerabilities == 'true'
         with:
           scan-type: 'fs'
-          format: 'table'
+          format: 'sarif'
           exit-code: '0'
           ignore-unfixed: true
-          severity: 'CRITICAL,HIGH'
+          severity: 'CRITICAL'
+          output: 'trivy-results.sarif'
           hide-progress: true
 
-      - name: Run Trivy vulnerability scanner
+      - name: Upload Trivy scan report to GitHub Security tab
+        uses: github/codeql-action/upload-sarif@v2
+        if: inputs.publish_vulnerabilities == 'true'
+        with:
+          sarif_file: 'trivy-results.sarif'
+
+      - name: Test with Trivy vulnerability scanner
         uses: aquasecurity/trivy-action@master
         with:
           scan-type: 'fs'
-          format: 'sarif'
+          format: 'table'
           exit-code: '1'
           ignore-unfixed: true
           severity: 'CRITICAL'
-          output: 'trivy-results.sarif'
           hide-progress: true
 
-      - name: Upload Trivy scan results to GitHub Security tab
```

Comment:
Didn't mean to remove this completely, just remove the if: always() line

Reply:
Oh it just got moved around. The `git diff` output may be a bit confusing, but here's the new code snippet:

```
      - name: Generate Trivy vulnerability scan report
        uses: aquasecurity/trivy-action@master
        if: inputs.publish_vulnerabilities
        with:
          scan-type: 'fs'
          format: 'sarif'
          exit-code: '0'
          ignore-unfixed: true
          severity: 'CRITICAL'
          output: 'trivy-results.sarif'
          hide-progress: true

      - name: Upload Trivy scan report to GitHub Security tab
        uses: github/codeql-action/upload-sarif@v2
        if: inputs.publish_vulnerabilities
        with:
          sarif_file: 'trivy-results.sarif'

      - name: Test with Trivy vulnerability scanner
        uses: aquasecurity/trivy-action@master
        with:
          scan-type: 'fs'
          format: 'table'
          exit-code: '1'
          ignore-unfixed: true
          severity: 'CRITICAL'
          hide-progress: true
```